client/src/hooks/members/useDepartments.ts

```diff
@@ -0,0 +1,104 @@
+import { useMutation, useQuery, useQueryClient } from "@tanstack/react-query";
+import { DepartmentDetailsDto, DepartmentMemberDto, DirectorDto } from "../../types/membersTypes";
+import { useContext } from "react";
+import { AuthContext } from "../../context/auth-context/AuthContext";
+import { AxiosError } from "axios";
+import {
+  getAllDirectors,
+  getCurrentDirectors,
+  getDepartmentMembers,
+  getDepartments,
+  updateDepartment,
+} from "../../api/departments";
+import { authReducerActionType } from "../../types/globalTypes";
+import { showErrorMessage, showSuccessMessage } from "../../utils/toastUtils";
+
+/**
+ * Hook that handles the department api calls, uses react-query
+ * @returns The departments, the department members, the current directors, all directors and a function to update the details of a department
+ */
+const useDepartments = () => {
+  const { dispatchAuth } = useContext(AuthContext);
+  const queryClient = useQueryClient();
+
+  // ############
+  // GET QUERIES
+  // ############
+
+  // getDepartments query
+  const { data: departmentsData } = useQuery({
+    queryKey: ["departments"],
+    queryFn: getDepartments,
+  });
+
+  const departments = (departmentsData?.data as DepartmentDetailsDto[]) || [];
```

Comment:
This can be applied to all hook data: Would this be better for readability?
```suggestion
  const departments = departmentsData?.data || [];
```